### PR TITLE
Use consistent colors on transaction page when disputed.

### LINKED ISSUE #7396: Use DisputeStatusChip on transaction details page
### Description

`<PaymentStatusChip />` is [used](https://github.com/Automattic/woocommerce-payments/blob/fc6264b1ecd23b22b690e3cc702ff3fc7fa8c786/client/payment-details/summary/index.tsx#L237-L242) on the Payment Details screen. When a transaction is disputed `<DisputeStatusChip />` should be used instead for consistency.

`<DisputeStatusChip />` has a color change based on urgency that is shown in the other screens. This same component should be used on the transaction details page as well. This will mean that the color of the background notice and the StatusChip are the same,

Image

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #7396 

#### Changes proposed in this Pull Request

This PR replaces the `<PaymentStatusChip />` with `<DisputeStatusChip />` when a transaction is disputed.

This allows for the urgency of the dispute to factor into the color of the chip which will match with the banner notice on the page.

**Before:**
image

**After:**
image

#### Testing instructions

* Create a disputed order
* In `client/disputes/utils.ts` update `isDueWithin` to always return `true`
* View the transaction details page of a disputed order.
* Confirm the StatusChip matches the BannerNotice color

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/components/dispute-status-chip/index.tsx [hunk 1]
```diff
@@ -21,10 +22,24 @@ import type {
...
+	let message = mapping.message || formatStringValue( status );
+
+	// Statuses starting with warning_ are Inquiries and these are already prefaced with "Inquiry: "
+	if ( includeTransactionType && ! status.startsWith( 'warning' ) ) {
+		message = sprintf(
+			/** translators: %s is the status of the Dispute. */
+			__( 'Disputed: %s', 'woocommerce-payments' ),
+			message
+		);
+	}
```
Comment (brucealdridge) [inline client/components/dispute-status-chip/index.tsx]: I'm not super confident about this approach, seems a little hacky. Also open to suggestions on a better flag that `includeTransactionType`

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7398` or branch name `fix/7396-disputestatuschip` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7396-disputestatuschip) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 13d4f3a06073d64a443ddca4e143f4c1b9426bbe
- Build time: 2023-10-06 04:51:32 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/dispute-status-chip/index.tsx (hunk 1 above)
Comment (haszari) [reply client/components/dispute-status-chip/index.tsx]: Can we just pass in the message from above? This thing doesn't need to format the message (much).

File: client/components/dispute-status-chip/index.tsx [hunk 2]
```diff
@@ -21,10 +22,24 @@ import type {
 interface Props {
 	status: DisputeStatus | string;
 	dueBy?: CachedDispute[ 'due_by' ] | EvidenceDetails[ 'due_by' ];
+	includeTransactionType?: boolean;
```
Comment (Jinksi) [inline client/components/dispute-status-chip/index.tsx]: There might be an opportunity to rename this more specifically to something like `prefixDisputeType` but we can follow this up if/when we decide to refactor.

File: client/components/dispute-status-chip/index.tsx (hunk 1 above)
Comment (brucealdridge) [reply client/components/dispute-status-chip/index.tsx]: Unfortunately not as this message is used to map to different statuses and colors

**Review by Jinksi — APPROVED** [review]
This works well @brucealdridge 

We have had some discussions on Slack about the _best_ way to disentangle the string mapping and potentially create a simpler API for this, but I think we should ship this improvement to our merchants first and consider a refactor for a future release.

Disputed: Needs response within 1 day ✅

image

Disputed: Needs response within 7 days ✅
image

Inquiry: Needs response within 1 day ✅
image


Inquiry: Needs response within 7 days ✅
image

In addition, I checked other transaction statuses and all rendered as expected: Paid, refunded, disputes under review, won, lost, and inquiry closed/under review.

File: client/components/dispute-status-chip/index.tsx (hunk 1 above)
Comment (haszari) [reply client/components/dispute-status-chip/index.tsx]: Yea to make this component dumb we need to do some brain surgery / lobotomy 🔪

Comment (haszari) [reply client/components/dispute-status-chip/index.tsx]: The colour / urgency would be an excellent prop to expose 😁

**Pull request merged**